modules/gdscript/editor/script_templates/CharacterBody3D/basic_movement.gd

```diff
@@ -5,29 +5,29 @@ extends _BASE_
 
 const SPEED = 5.0
 const JUMP_VELOCITY = 4.5
+const FRICTION = 10.0
 
-# Get the gravity from the project settings to be synced with RigidBody nodes.
+# Get the gravity from the project settings, to allow syncing with RigidBody3D nodes.
 var gravity: float = ProjectSettings.get_setting("physics/3d/default_gravity")
 
 
 func _physics_process(delta: float) -> void:
 	# Add the gravity.
-	if not is_on_floor():
-		velocity.y -= gravity * delta
+	velocity.y -= gravity * delta
 
-	# Handle Jump.
+	# Allow jumping only when on the floor.
 	if Input.is_action_just_pressed("ui_accept") and is_on_floor():
 		velocity.y = JUMP_VELOCITY
 
 	# Get the input direction and handle the movement/deceleration.
-	# As good practice, you should replace UI actions with custom gameplay actions.
+	# As good practice, you should replace UI actions with custom input actions.
 	var input_dir := Input.get_vector("ui_left", "ui_right", "ui_up", "ui_down")
 	var direction := (transform.basis * Vector3(input_dir.x, 0, input_dir.y)).normalized()
 	if direction:
 		velocity.x = direction.x * SPEED
 		velocity.z = direction.z * SPEED
 	else:
-		velocity.x = move_toward(velocity.x, 0, SPEED)
-		velocity.z = move_toward(velocity.z, 0, SPEED)
+		velocity.x = move_toward(velocity.x, 0, FRICTION * delta)
+		velocity.z = move_toward(velocity.z, 0, FRICTION * delta)
```

Comment:
I gave this a test, and actually this is vastly worse. The character slips and slides like crazy.

I'm thinking that it would be better to just simplify this script and make it like this:

```gdscript
	# Get the input direction and handle the movement/deceleration.
	# As good practice, you should replace UI actions with custom input actions.
	var input_dir = Input.get_vector("ui_left", "ui_right", "ui_up", "ui_down")
	var direction = (transform.basis * Vector3(input_dir.x, 0.0, input_dir.y)).normalized()
	velocity.x = move_toward(velocity.x, direction.x * SPEED, delta * 50.0)
	velocity.z = move_toward(velocity.z, direction.z * SPEED, delta * 50.0)
```

CC @fabriceci, what was the intended behavior here? I don't see why we need an `if` statement.

Reply:
You felt it was crazy? The math works out to about a 0.5 second deceleration. Enough to show it's got some but not too slippery.

Using `50.0` results in such high accel/decel that it might as well not be there (it's about 6 frames). At that point we might as well just set the speed and keep the whole thing simpler:

```gdscript
velocity.x = direction.x * SPEED
velocity.z = direction.z * SPEED
```

Reply:
Good point, I think just doing `velocity.x = direction.x * SPEED` would be even better.

Reply:
OK. Se we have a basic movement script and people can add accel/decel on their own. It's probably better to keep it all simple.